### ISSUE TITLE
Building Issues/Detail the Build Environment?

### ISSUE THREAD

**Issue by windowshopr** [Accenture/Spartacus#4]
Just came across your repo and am eager to get started using it, however I'm a bit of a n00b when it comes to .NET projects, and it seems like every time I try to load and build ANY .NET project in VS, it's just rot with red underlines and version complaints, very frustrating.

That being said, I am currently running Visual Studio Community 2022, latest update was just done today, 17.7.4, on Windows 10.

When I first opened the project, I got these error messages:

![image](https://github.com/Accenture/Spartacus/assets/40477419/8ff2519c-ae7b-415b-81df-00601746112c)

After some reading, I discovered that the likely cause was due to my .NET version not being high enough, as denoted from this table:

```
╔══════════════════╦═════════╦═════════════════════════════╗
║ Target framework ║ version ║ C# language version default ║
╠══════════════════╬═════════╬═════════════════════════════╣
║ .NET             ║ 6.x     ║ C# 10                       ║
║ .NET             ║ 5.x     ║ C# 9.0                      ║
║ .NET Core        ║ 3.x     ║ C# 8.0                      ║
║ .NET Core        ║ 2.x     ║ C# 7.3                      ║
║ .NET Standard    ║ 2.1     ║ C# 8.0                      ║
║ .NET Standard    ║ 2.0     ║ C# 7.3                      ║
║ .NET Standard    ║ 1.x     ║ C# 7.3                      ║
║ .NET Framework   ║ all     ║ C# 7.3                      ║
╚══════════════════╩═════════╩═════════════════════════════╝
```

I think I had a .NET Framework installed, and not just .NET,

So in the VS Installer, I uninstalled basically everything .NET related to attempt a fresh start. I open the solution file again, and am greeted with this message:

![image](https://github.com/Accenture/Spartacus/assets/40477419/c2474ea6-1d95-4f8e-9685-96a7be91574d)

So, I do the Recommended step and update the target .NET Framework 4.8, where I now see this error:

![image](https://github.com/Accenture/Spartacus/assets/40477419/9683afd1-4f19-48a9-9b69-e2b4e0fdaa29)

So again, I follow the steps and navigate to `https://aka.ms/msbuild/developerpacks`, and download the .NET Framework 4.8 Developer Pack, and run the installer.

After reloading the solution file in VS, I get the same group of errors as before:

![image](https://github.com/Accenture/Spartacus/assets/40477419/94dade00-a4de-4681-85ce-4df8e41f4e54)

Wonderful, stuck in a loop!

After reading a few forum posts, one answer suggested to review what was in the `right click the project (not solution) in Solution Explorer window > Properties > Build > Advanced button at the bottom right > Language version`, which mine shows as:

![image](https://github.com/Accenture/Spartacus/assets/40477419/3b2805bd-80a1-43c3-80bd-dbef69e88d85)

Following the link on "Why can't I select a different C# version" just takes you to [this page](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version) just highlighting the map between .NET versions and C# language versions.

On that page, and another forum post, they suggested to change the `<LangVersion>' parameter in the `.csproj` file of the project to whatever version you want to "target", but that wasn't helpful as the `<LangVersion>` in Spartacus' file is already set to version 9.0:

![image](https://github.com/Accenture/Spartacus/assets/40477419/fe70ca4e-033d-4083-b2de-7d26be71e389)

They say changing that value from `9.0` to `preview` or even just getting rid of the line entirely forces it to target the latest language available for your .NET version, but that didn't work either.

So if I had to guess, it seems like I'm somehow "stuck" on C# version 7.3 and have exhausted the resources I've attempted to fix it. Forgive me if this is a simple solution, but as you can see, I've put as much effort in as I know how, and while this isn't a Spartacus issue, I can't help but think it would benefit someone also in my boat in the future to figure this out.

What can I try to get this working?

**Comment by windowshopr** [comment]
I should note that I have also tried uninstalling .NET Framework 4.8, and installed .NET 8.0 SDK separately and on its own. When I load the solution file again, I get the same message re: "Update the target to .NET Framework 4.8 (Recommended), and that didn't work either.

**Comment by sadreck** [comment]
Hello there,

That's one detailed issue! I started googling as well and ended up at the same conclusion as you - "it depends on the version of .NET", but if you installed .NET 8 then it should have definitely worked. I'm afraid I can't help with solving the VS IDE issue any more than suggesting "have you tried uninstalling the entire Visual Studio and installing it again?" and "try installing VS inside a VM and see what happens".

Other than that, you could use the pre-compiled version that's under releases - https://github.com/Accenture/Spartacus/releases

Pavel

**Comment by windowshopr** [comment]
right!? I will definitely try a clean uninstall/reinstall of VS. Hate doing that, but will attempt it.

Just as you sent your message I saw the .zip file with the precompiled exe in it! So I will use that to start using the program, and will post any results from reinstalling VS. Thanks!

**Comment by windowshopr** [comment]
Yeah tried uninstalling VS, and I noticed that there were some independent installations of some old .NET stuff in my applications list for uninstalling:

![image](https://github.com/Accenture/Spartacus/assets/40477419/881de3c0-5d29-4c1d-9612-51e0bbbe7730)

So I uninstalled all of those as well. Reinstalled VS 2022 with NO packs. Then opened up the solution file, and clicked the "install missing packages for a better development experience" or whatever the link was, thinking that VS would install what was necessary for the solution. After all that, same issue re: "xxxx is not available in C# 7.3. Please use language version 9.0 or greater", WHICH I AM ACCORDING TO WHAT IS INSTALLED FROM WHAT I CAN TELL!!! >:( so I'm at a loss. Thanks for providing a precompiled solution :'D
